revision.c

```diff
@@ -2334,7 +2335,7 @@ static int handle_revision_opt(struct rev_info *revs, int argc, const char **arg
 		revs->ignore_missing = 1;
 	} else if (opt && opt->allow_exclude_promisor_objects &&
 		   !strcmp(arg, "--exclude-promisor-objects")) {
-		if (fetch_if_missing)
+		if (repo.fetch_if_missing)
```

Comment:
IIRC `struct rev_info` contains a pointer to the repository in question. `repo` does not exist at this point.